src/Simulation.jl

```diff
@@ -0,0 +1,84 @@
+module Simulation
+
+export Simulation, run!
+
+using Oceananigans.Models
+using Oceananigans.Utils
+
+struct Simulation{M, T, SI, ST, W, R, D, O, P}
```

Comment:
This looks really great. Should we make it mutable for convenience? Some people may want a constant time-step, but still have the ability to change it during the simulation. Why is runtime one word?

Can also add a Boolean “.stop” which turns from false to true when a stop criteria is reached. With that we can work on a functional syntax that causes simulation stoppage, and allow users to write their own stop criteria via functions.